modules/darwin/src/main/ts/ephox/darwin/selection/Selections.ts

```diff
@@ -0,0 +1,20 @@
+import { Optional } from '@ephox/katamari';
+import { SugarElement } from '@ephox/sugar';
+import * as TableSelection from '../api/TableSelection';
+import * as SelectionTypes from './SelectionTypes';
+
+export interface Selections {
+  get: () => SelectionTypes.SelectionType;
+}
+
+// tslint:disable-next-line:variable-name
+export const Selections = (lazyRoot: () => SugarElement<Element>, getStart: () => Optional<SugarElement<HTMLTableCellElement | HTMLTableCaptionElement>>, selectedSelector: string): Selections => {
+  const get = () => TableSelection.retrieve(lazyRoot(), selectedSelector).fold(
+    () => getStart().map(SelectionTypes.single).getOr(SelectionTypes.none()),
```

Comment:
`getOrThunk(SelectionTypes.none)` would be better here, as that way it's not eagerly creating the none selection type object.